src/context/meta-tags-context.php

```diff
@@ -655,6 +674,40 @@ public function generate_main_image_id() {
 		}
 	}
 
+	/**
+	 * Gets the main image ID for REST requests.
+	 *
+	 * @return int|null The main image ID.
+	 */
+	protected function get_main_image_id_for_rest_request() {
+		switch ( $this->page_type ) {
+			case 'Post_Type':
+				if ( $this->post instanceof WP_Post ) {
+					return $this->get_singular_post_image( $this->post->ID );
```

Comment:
It is a bit strange that the seemingly only difference is where we get the ID from.
I suppose the tricky thing is in the extra safety checks too, like the `instanceof WP_Post` here, and the `$GLOBALS['wp_query']` for non-single. Making it hard to just get an ID depending on the context and then do the same thing.

Reply:
Yeah, I'd prefer no to touch the normal flow